src/pages/api/thirdparty/graphql.tsx

```diff
@@ -42,6 +61,39 @@ export default startServerAndCreateNextHandler(server, {
     try {
       const client = await clientConnection()
       const mongodb = client.db(process.env.MONGODB_DB)
+
+      const foundUser = await User.findOne(userId, { db: mongodb })
+
+      if (!foundUser) {
+        // Authenticated user not found in portal db
+        // Create new record for them
+        try {
+          // Create new user in portal db
+          const initCollection = await getExampleCollection()
+
+          await User.createOne(
+            userId,
+            [initCollection],
+            // Default display name is the user's CN since we do not
+            // have their first and last name on the token.
+            // If/when they log in to the portal, they can update
+            userId,
+            'light',
+            {
+              db: mongodb,
+            }
+            // Q: Should we include last login? They are technically
+            // not logging in to the portal. Do we want to use it to track
+            // API access as well?
```

Comment:
Question here for @gidjin @jcbcapps @jbecker01  -- do we want to track `lastLogIn` field for a user created via third-party API?